compiler/noirc_evaluator/src/ssa/checks/check_for_underconstrained_values.rs

```diff
@@ -327,7 +326,27 @@ impl Context {
             }
             updated_sets.insert(largest_set_index, largest_set);
         }
-        self.value_sets = updated_sets.values().cloned().collect();
+        updated_sets.values().cloned().collect()
+    }
+
+    /// Parallel version of merge_sets
+    /// The sets are merged by chunks, and then the chunks are merged together
+    fn merge_sets_par(sets: &[HashSet<ValueId>]) -> Vec<HashSet<ValueId>> {
+        let mut sets = sets.to_owned();
+        let mut len = sets.len();
+        let mut prev_len = len + 1;
+
+        while len > 1000 && len < prev_len {
+            let chunks: Vec<_> = sets.par_chunks(1000).collect();
```

Comment:
```suggestion
        while len > 1000 && len < prev_len {
            let chunks: Vec<_> = sets.par_chunks(1000).collect();
```

Do we need the `len > 1000` condition? `par_chunks` should work with fewer chunks afaik. Do we end up slower with the parallelisation overhead without this?

Reply:
Yes I think we need the condition. The point is that we need to do a (sequential) merge_sets at the end, because the chunks produce sets that are properly merged inside the chunk, but they may intersect with sets from other chunks. Testing the len is a simple way to know if we have chunks or not.

Reply:
Sounds good.